### PR TITLE
Add block mobile toolbar with move/settings/delete controls.

### LINKED ISSUE #705: Mobile: Show mover in block-toolbar on mobile
We need to surface the controls for up, down, trash and advanced on mobile. Let's also surface them on wide images. 

Here's how it looks on desktop now:

block ui desktop

On "wide" images we move the controls above the image. Let's do the same for the mover:

block ui wide

On mobile, let's try a new interface. Let's have an ellipsis on the right of the selected block:

block ui mobile collapsed

When you click this ellipsis, a new toolbar expands and replaces the old one:

block ui mobile expanded

We can do this with simple responsive CSS. That is, we always have the mover, trash and cog present in the above-the-block toolbar, and simply hide these on desktop breakpoints, show them on mobile. I've checked the accessibility implications with @afercia, and so long as the controls are `display: none;` when they aren't used, they aren't announced by screenreaders or show up in the tab stack. On mobile, there is no tabbing, so the requirements here are different.

[Older mockup can be seen here](https://cloud.githubusercontent.com/assets/1204802/25796649/9eae74f6-33da-11e7-879a-08733e414ec5.png).

### PR DESCRIPTION
**Pull request by mtias**
https://cloudup.com/cBIktw2nLPE

Closes #705.

### CODE REVIEW

**Comment by mtias** [comment]
@jasmussen going with this icon for now because we don't have an ellipsis.

**Comment by jasmussen** [comment]
We should have. Try it, `ellipsis`. Our Dashicons sprite is ahead of the dev docs.

**Comment by jasmussen** [comment]
Btw 😍

File: editor/modes/visual-editor/style.scss [hunk 1]
```diff
@@ -65,6 +65,34 @@
 		transition: 0.2s outline;
 	}
 
+	&.is-showing-mobile-controls {
+		.editor-visual-editor__block-controls > div {
```
Comment (aduth) [inline editor/modes/visual-editor/style.scss]: Relies on slot behavior wrapping with `div`. Might be less fragile by targeting anything _except_ mobile tools:

```css
.editor-visual-editor__block-controls > :not( .editor-visual-editor__mobile-tools ) {
```

File: editor/modes/visual-editor/style.scss [hunk 2]
```diff
@@ -65,6 +65,34 @@
...
+			.editor-block-settings-menu {
+				display: block;
+				display: flex;
+				position: static;
+				.components-button {
+					margin: 0 5px;
+				}
+			}
+			.editor-block-settings-menu {
+				z-index: 99;
```
Comment (aduth) [inline editor/modes/visual-editor/style.scss]: Should use `z-index` function.

https://github.com/WordPress/gutenberg/blob/master/editor/assets/stylesheets/_z-index.scss

File: editor/modes/visual-editor/block.js [hunk 3]
```diff
@@ -364,6 +376,19 @@ class VisualEditorBlock extends Component {
 						<div className="editor-visual-editor__block-controls">
 							<BlockSwitcher uid={ block.uid } />
 							<Slot name="Formatting.Toolbar" />
+							<Toolbar className="editor-visual-editor__mobile-tools">
+								{ ( showUI || isHovered ) && <BlockMover uids={ [ block.uid ] } /> }
+								{ ( showUI || isHovered ) && <BlockRightMenu uid={ block.uid } /> }
+								<IconButton
+									className="editor-visual-editor__mobile-toggle"
+									onClick={ this.toggleMobileControls }
+									aria-label={ __( 'Toggle extra block controls' ) }
```
Comment (aduth) [inline editor/modes/visual-editor/block.js]: Redundant with `label` prop below (`IconButton` uses `label` to apply `aria-label`).

File: editor/modes/visual-editor/block.js [hunk 4]
```diff
@@ -364,6 +376,19 @@ class VisualEditorBlock extends Component {
...
+									aria-expanded={ true }
```
Comment (aduth) [inline editor/modes/visual-editor/block.js]: Minor: `={ true }` can be dropped.

File: editor/modes/visual-editor/block.js [hunk 5]
```diff
@@ -364,6 +376,19 @@ class VisualEditorBlock extends Component {
...
+									label={ __( 'Toggle Controls' ) }
+								>
+									<Dashicon icon="ellipsis" />
```
Comment (aduth) [inline editor/modes/visual-editor/block.js]: `IconButton` renders `Dashicon` via its `icon` prop, i.e. `<IconButton icon="ellipsis" />`

File: editor/modes/visual-editor/block.js [hunk 6]
```diff
@@ -364,6 +376,19 @@ class VisualEditorBlock extends Component {
 						<div className="editor-visual-editor__block-controls">
 							<BlockSwitcher uid={ block.uid } />
 							<Slot name="Formatting.Toolbar" />
+							<Toolbar className="editor-visual-editor__mobile-tools">
+								{ ( showUI || isHovered ) && <BlockMover uids={ [ block.uid ] } /> }
```
Comment (aduth) [inline editor/modes/visual-editor/block.js]: Curious if we need both BlockMover here and above, or if we could apply styles to the root element when `'is-showing-mobile-controls'` is applied to appear as it does with the `Toolbar` component.

Comment (mtias) [inline editor/modes/visual-editor/block.js]: Tried this but could not get it to work on mobile reliably between proper flex positioning, sticky position, etc.

File: editor/modes/visual-editor/style.scss (hunk 2 above)
Comment (mtias) [inline editor/modes/visual-editor/style.scss]: It doesn't seem to be needed anymore with the latest approach.

**Comment by mtias** [comment]
Going to merge this.

**Pull request merged**